nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/autodiff/samediff/optimize/optimizations/CuDNNFunctionOptimizations.java

```diff
@@ -0,0 +1,112 @@
+/*
+ *  ******************************************************************************
+ *  *
+ *  *
+ *  * This program and the accompanying materials are made available under the
+ *  * terms of the Apache License, Version 2.0 which is available at
+ *  * https://www.apache.org/licenses/LICENSE-2.0.
+ *  *
+ *  *  See the NOTICE file distributed with this work for additional
+ *  *  information regarding copyright ownership.
+ *  * Unless required by applicable law or agreed to in writing, software
+ *  * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *  * License for the specific language governing permissions and limitations
+ *  * under the License.
+ *  *
+ *  * SPDX-License-Identifier: Apache-2.0
+ *  *****************************************************************************
+ */
+
+package org.nd4j.autodiff.samediff.optimize.optimizations;
+
+import org.nd4j.autodiff.samediff.ArrayHolder;
+import org.nd4j.autodiff.samediff.SDVariable;
+import org.nd4j.autodiff.samediff.SameDiff;
+import org.nd4j.autodiff.samediff.internal.SameDiffOp;
+import org.nd4j.autodiff.samediff.optimize.OptimizationHelper;
+import org.nd4j.autodiff.samediff.optimize.Optimizer;
+import org.nd4j.linalg.api.ops.impl.layers.convolution.Conv2D;
+import org.nd4j.linalg.factory.Nd4j;
+
+import java.util.List;
+
+public class CuDNNFunctionOptimizations extends BaseOptimizerSet {
+
+    protected static final boolean isCudaBackend;
+
+    static {
+        String backend = Nd4j.getExecutioner().getEnvironmentInformation().getProperty("backend");
+//        isCudaBackend = "CUDA".equalsIgnoreCase(backend);
+        isCudaBackend = true;   //For testing only
```

Comment:
If that is for testing only, I think it shouldn't make it into a PR.

Reply:
Fixed.